lib/internal/async_hooks.js

```diff
@@ -250,7 +252,7 @@ function newUid() {
 }
 
 function getOrSetAsyncId(object) {
-  if (object.hasOwnProperty(async_id_symbol)) {
+  if (ReflectHas(object, async_id_symbol)) {
```

Comment:
This isn't semantically equivalent. It includes the prototype properties.

Reply:
are there random symbols on object prototype chains?

Reply:
It's more that if we don't care about prototype then we should just have this be an `undefined` check. I assume this was done for a specific reason but I wasn't the one that wrote that code so I don't know...

Reply:
Let's not change the semantics here. You just need to set the prototype of the object to another object that already has an async id to make this behave in an unintended way.